src/main/webapp/faceted-dropdown.tsx

```diff
@@ -0,0 +1,126 @@
+import * as React from 'react'
+import gql from 'graphql-tag'
+import { useQuery } from '@apollo/react-hooks'
+import LinearProgress from '@material-ui/core/LinearProgress'
+import Box from '@material-ui/core/Box'
+import { getIn } from 'immutable'
+import Select, { SelectProps } from '@material-ui/core/Select'
+import MenuItem from '@material-ui/core/MenuItem'
+import Checkbox from '@material-ui/core/Checkbox'
+import ListItemText from '@material-ui/core/ListItemText'
+import FormControl from '@material-ui/core/FormControl'
+import InputLabel from '@material-ui/core/InputLabel'
+import { FormHelperText } from '@material-ui/core'
+const { useApolloFallback } = require('./react-hooks')
+
+const QUERY = gql`
+  query getOptions($attribute: String!) {
+    facet(attribute: $attribute) {
+      value
+    }
+    metacardTypes {
+      id
+      enums
+    }
+  }
+`
+
+type Props = SelectProps & {
+  facetAttribute: string // Attribute to facet against for values. NOTE: attributes need to be **whitelisted by an Admin** before they can be faceted
+  noOptionsText?: string
+  renderMenuItem?: any
+  label?: string
+  error?: string
+}
+
+type DropdownProps = Props & {
+  options?: string[]
+}
+
+const FacetedDropdown = (props: Props) => {
+  const { facetAttribute } = props
+  const { data, loading } = useQuery(QUERY, {
+    variables: { attribute: facetAttribute },
+  })
+
+  if (loading) {
+    return <LinearProgress />
+  }
+
+  const facetEnums = getIn(data, ['facet'], []).map((facet: any) => facet.value)
+  let enums = facetEnums
+  const attributeDefinitions = getIn(data, ['metacardTypes'], [])
+  if (enums.length === 0) {
+    enums =
+      getIn(
+        attributeDefinitions.find(
+          (definition: any) => definition.id === facetAttribute
+        ),
+        ['enums'],
+        undefined
+      ) || []
+  }
+
+  return <Dropdown {...props} options={enums} />
+}
+
+const autogenerateOptions = (facet: string) => {
```

Comment:
Idk if there's a better way to handle storybook than this. This is what causes the difference in screenshots